### DISCUSSION TITLE
Could we memoise the bind function?

### DISCUSSION THREAD

**Discussion by adamledwards** [pmndrs/use-gesture#481] [Ideas]
Hello first of all this is such a great library, thank you. 
I also wanted to see if it was a feasible idea to hold memoise the function created by Function.prototype.bind. 

My reason is it could be used by apps to use the memo function in their components.

https://github.com/pmndrs/use-gesture/blob/dc731a080ec20fa07a2891e59eb62494e7276066/packages/react/src/useRecognizers.ts#L45-L47

In the snipped below, if the bind function was true in a strict equality test, the GestureComponent could skip the rerender. 
This would be useful in a large list of elements that use gesture event handlers.

```jsx
import {memo, useState} from 'react'

const GestureComponent = memo(({bind, argument}) => {
    <div {...bind(argument)}>My Gesture component</div>
})

function App() {
    const bind = useXXXX(() => {})
    const [counter, increment] = useState(0)
    return (
        <div>
            <button onClick={() => increment(c => c + 1)}>counter {counter}</button>
            {list.map((key) => <GestureComponent key={key} bind={bind} argument="some argument"/>)
        </div>
    )
}
```

**Comment by dbismut** [comment]
It's been a long time, I can re-investigate why I made that choice. I know I tried to memo at some point but it had some side effects. The first one being that it would freeze state values, although we could add a third dependency argument to the hook. That would be a breaking change though. But I think there were some other collaterals involving config, I'd have to investigate.

**Reply by adamledwards** [reply]
Thank you. I'll also give it a fork and see check out the side effects. When I have a moment this weekend
